### PR TITLE
aws_vpc: accept 17 hexadecimal characters for vpc_id

### LINKED ISSUE #3518: aws_vpc expect handle only 8 hexadecimal characters instead both 8 and 17
Description:
InSpec doesn't handle vpc_id which contain 17 characters and expect just 8

Case:
describe aws_subnets.where(vpc_id: 'vpc-012345678987654321') do
  it { should exist }

Expected result:
✔   aws_subnets exist
Actual result:
× The vpc ID 'vpc-01234567898765432' does not exist
VPC_ID should be in the format "vpc-" followed by 8 or 17 hexadecimal characters.

inspec version:
2.3.24

### PR DESCRIPTION
**Pull request by kchistova**
Signed-off-by kchistova <kchistova@gmail.com>

Fixes #3518 	
EDIT: (@clintoncwolfe) - move issue ref to PR desc

### CODE REVIEW

**Comment by clintoncwolfe** [comment]
Thanks @kchistova! Could you take a moment and sign your commits using `git commit -s`?  We use commit signing to indicate that the contributor has read and understood the [Developer Certification of Origin](https://github.com/inspec/inspec/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco), which basically means it's your code you're contributing, it doesn't introduce a new license, etc.  Thanks!

**Comment by clintoncwolfe** [comment]
Hrm, I think the DCO detector issued a false negative here - the commit is signed-off.

File: lib/resources/aws/aws_vpc.rb [hunk 1]
```diff
@@ -30,8 +30,8 @@ def validate_params(raw_params)
       allowed_scalar_type: String,
     )
 
-    if validated_params.key?(:vpc_id) && validated_params[:vpc_id] !~ /^vpc\-[0-9a-f]{8}/
-      raise ArgumentError, 'aws_vpc VPC ID must be in the format "vpc-" followed by 8 hexadecimal characters.'
+    if validated_params.key?(:vpc_id) && validated_params[:vpc_id] !~ /^vpc\-([0-9a-f]{8})|(^rtb\-[0-9a-f]{17})$/
```
Comment (clintoncwolfe) [inline lib/resources/aws/aws_vpc.rb]: `rtb-` is the prefix for *routing tables*, I think - don't you mean `vpc-` in both places?

File: test/unit/resources/aws_vpc_test.rb [hunk 2]
```diff
@@ -16,14 +16,22 @@ def test_empty_params_ok
     AwsVpc.new
   end
 
-  def test_accepts_vpc_id_as_scalar
+  def test_accepts_vpc_id_as_scalar_eight_sigh
```
Comment (clintoncwolfe) [inline test/unit/resources/aws_vpc_test.rb]: What does `eight_sigh` mean?

File: test/unit/resources/aws_vpc_test.rb [hunk 3]
```diff
@@ -16,14 +16,22 @@ def test_empty_params_ok
     AwsVpc.new
```
Comment (clintoncwolfe) [inline test/unit/resources/aws_vpc_test.rb:16]: Could you also update the examples in docs/resources/aws_vpc.md.erb?

**Review by clintoncwolfe — CHANGES_REQUESTED** [review]
Thanks for this - it really helps. 

A few things, all minor - only the regex change is a real blocker.  Thanks!

File: lib/resources/aws/aws_vpc.rb (hunk 1 above)
Comment (kchistova) [reply lib/resources/aws/aws_vpc.rb]: Yes, sorry

File: test/unit/resources/aws_vpc_test.rb (hunk 2 above)
Comment (kchistova) [reply test/unit/resources/aws_vpc_test.rb]: 8 sign i meant

File: lib/resources/aws/aws_vpc.rb (hunk 1 above)
Comment (kchistova) [reply lib/resources/aws/aws_vpc.rb]: Fixed

File: test/unit/resources/aws_vpc_test.rb (hunk 2 above)
Comment (kchistova) [reply test/unit/resources/aws_vpc_test.rb]: Changed to eight_sign

File: test/unit/resources/aws_vpc_test.rb (hunk 3 above)
Comment (kchistova) [reply test/unit/resources/aws_vpc_test.rb:16]: Done

**Review by clintoncwolfe — APPROVED** [review]

**Review by jquick — APPROVED** [review]
Looks good to me! Thanks @kchistova

**Pull request merged**